src/BuildScriptGeneratorCli/Commands/DockerfileCommand.cs

```diff
@@ -90,37 +90,33 @@ internal override int Execute(IServiceProvider serviceProvider, IConsole console
 
             int exitCode;
             var logger = serviceProvider.GetRequiredService<ILogger<DockerfileCommand>>();
-            using (var timedEvent = logger.LogTimedEvent("DockerFileCommand", buildEventProps))
+
+            ILoggerFactory loggerFactory = serviceProvider.GetRequiredService<ILoggerFactory>();
+            var sourceRepo = new LocalSourceRepo(this.SourceDir, loggerFactory);
+            var ctx = new DockerfileContext
             {
-                ILoggerFactory loggerFactory = serviceProvider.GetRequiredService<ILoggerFactory>();
-                var sourceRepo = new LocalSourceRepo(this.SourceDir, loggerFactory);
-                var ctx = new DockerfileContext
-                {
-                    SourceRepo = sourceRepo,
-                };
+                SourceRepo = sourceRepo,
+            };
 
-                var dockerfile = serviceProvider.GetRequiredService<IDockerfileGenerator>().GenerateDockerfile(ctx);
-                if (string.IsNullOrEmpty(dockerfile))
+            var dockerfile = serviceProvider.GetRequiredService<IDockerfileGenerator>().GenerateDockerfile(ctx);
+            if (string.IsNullOrEmpty(dockerfile))
+            {
+                exitCode = ProcessConstants.ExitFailure;
+                console.WriteErrorLine("Couldn't generate dockerfile.");
```

Comment:
Nit: please consider to directly return from this `if`, so that we can avoid the extra nesting coming from the `else` block below.